src/components/PluginLoader.jsx

```diff
@@ -97,6 +97,33 @@ const handleExternalNavigation = (iframeLoadEvent, pluginHref) => {
     }
 }
 
+/**
+ * Equivalent URLs:
+ * <appRoot>/
+ * <appRoot>/index.html
+ * <appRoot>/index.html?redirect=false
+ *
+ * newUrl is expected to come from pluginHref,
+ * and will always have /index.html?redirect=false
+ */
+const isBaseEquivalent = (currentUrlOrig, newUrlOrig) => {
+    // Copy before mutating
+    const [currentUrl, newUrl] = [new URL(currentUrlOrig), new URL(newUrlOrig)]
+    newUrl.searchParams.sort()
+    // If redirect is not already false in the search set that
+    currentUrl.searchParams.set('redirect', 'false')
+    currentUrl.searchParams.sort()
+    // If pathname doesn't end in html, add that
+    if (currentUrl.pathname.endsWith('/')) {
+        currentUrl.pathname = currentUrl.pathname + 'index.html'
+    }
+
+    const [newBase] = newUrl.href.split('#')
+    const [currentBase] = currentUrl.href.split('#')
+
+    return newBase === currentBase
```

Comment:
I think this might be more restrictive than it needs to be - we could probably just check for any sub-path of the base url - but I tested this and it works well!